File: web/tizen/src/SimidController.ts
import * as simid from '@broadpeak-tv/simid-controller'

export default class SimidController extends simid.SimidController {

  private _simidControllerApi: any /*GenericSimidControllerApi*/

  public set simidControllerApi(controllerApi: any) {
    this._simidControllerApi = controllerApi
  }

  protected receiveMessage(event: MessageEvent): void {
    this._simidControllerApi?.onMessageReceived(event.data)
    super.receiveMessage(event)
  }

  protected postMessage(message: simid.Message): void {
    this._simidControllerApi?.onMessageSent(JSON.stringify(message))
    super.postMessage(message)
  }
}

File: android/app/src/main/java/tv/broadpeak/simid/app/SimidController.kt
package tv.broadpeak.simid.app

import android.app.Activity
import android.content.Context
import android.graphics.Rect
import tv.broadpeak.simid.controller.Dimensions
import tv.broadpeak.simid.controller.CreativeData
import tv.broadpeak.smartlib.ad.simid.GenericSimidControllerApi

class SimidController(
    private val activity: Activity,
    private val context: Context,
    private val mainPlayerDimensions: Dimensions,
    private val creativeDimensions: Dimensions,
    private val creativeUri: String,
    private val creativeData: CreativeData,
    private val adDuration: Float = 0.0F,
    private val adSkippable: Boolean = false,
    private val mediaStatePollingInterval: Long = MEDIA_TIMEUPDATE_INTERVAL_MS
) : tv.broadpeak.simid.controller.SimidController(activity, context, mainPlayerDimensions, creativeDimensions, creativeUri, creativeData, adDuration, adSkippable, mediaStatePollingInterval) {

    private var simidControllerApi: GenericSimidControllerApi? = null

    fun simidControllerApi(controllerApi: GenericSimidControllerApi) {
        this.simidControllerApi = controllerApi
    }

    override fun receiveMessage(messageStr: String) {
        this.simidControllerApi?.onMessageReceived(messageStr)
        super.receiveMessage(messageStr)
    }

    override fun postMessage(message: String) {
        this.simidControllerApi?.onMessageSent(message)
        super.postMessage(message)
    }
}